mantle/cmd/ore/kubevirt/build.go

```diff
@@ -0,0 +1,104 @@
+package kubevirt
+
+import (
+	"archive/tar"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+)
+
+func BuildContainerDisk(imgPath string) (v1.Image, error) {
+	img := empty.Image
+	layerStream, errChan := StreamLayer(imgPath)
+	layer, err := tarball.LayerFromReader(layerStream)
+	if err != nil {
+		return nil, fmt.Errorf("error creating an image layer from disk: %v", err)
+	}
+
+	img, err = mutate.AppendLayers(img, layer)
+	if err != nil {
+		return nil, fmt.Errorf("error appending the image layer: %v", err)
+	}
+
+	if err := <-errChan; err != nil {
+		return nil, fmt.Errorf("error creating the tar file with the disk: %v", err)
+	}
+	return img, nil
+}
+
+func StreamLayer(imagePath string) (tarReader io.ReadCloser, errChan chan error) {
+	errChan = make(chan error, 1)
+	reader, writer := io.Pipe()
+	tarWriter := tar.NewWriter(writer)
+
+	go func() {
+		defer writer.Close()
+		defer tarWriter.Close()
+		err := addFileToTarWriter(imagePath, tarWriter)
+		if err != nil {
+			errChan <- fmt.Errorf("error adding file '%s', to tarball: %v", imagePath, err)
+		}
+		close(errChan)
+	}()
+
+	return reader, errChan
+}
+
+func addFileToTarWriter(filePath string, tarWriter *tar.Writer) error {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return fmt.Errorf("error opening file: %v", err)
+	}
+	defer file.Close()
+
+	stat, err := file.Stat()
+	if err != nil {
+		return fmt.Errorf("error getting file information with stat: %v", err)
+	}
+
+	header := &tar.Header{
+		Typeflag: tar.TypeDir,
+		Name:     "disk/",
+		Mode:     0555,
+		Uid:      107,
+		Gid:      107,
+		Uname:    "qemu",
+		Gname:    "qemu",
+		ModTime:  time.Now(),
+	}
+
+	err = tarWriter.WriteHeader(header)
+	if err != nil {
+		return fmt.Errorf("error writing disks directory tar header: %v", err)
+	}
+
+	header = &tar.Header{
+		Typeflag: tar.TypeReg,
+		Uid:      107,
```

Comment:
Does anything about containerdisk actually require the `qemu` user?  It seems oddly hardcoded. I don't think there's an equivalent of this on e.g. Debian systems is there?

Any reason not to just use 0/0?

Reply:
Funny enough `107:107` is what we need: https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#containerdisk-workflow-example. We do not really touch base OSes in kubevirt. libvirt comes with our kubevirt containers and is configured by our components to use `107` for the qemu user and have it available.

We have a RFE against libvirt open which will allow us to pass in file descriptors to not depend on this value in containerdisks (right now we do a bind-mount and we don't want to touch even the metadata of the file to be really sure that we don't accidentally trigger a layer copy).

Reply:
A link to that doc somewhere in this code would be good.

Reply:
Done.